horiz_interp/include/horiz_interp_conserve.inc

```diff
@@ -16,131 +16,22 @@
-  !!
-  !> @ingroup horiz_interp_conserve_mod
-  interface horiz_interp_conserve_new
-     module procedure horiz_interp_conserve_new_1dx1d
-     module procedure horiz_interp_conserve_new_1dx2d
-     module procedure horiz_interp_conserve_new_2dx1d
-     module procedure horiz_interp_conserve_new_2dx2d
-  end interface
-
-  !> @addtogroup horiz_interp_conserve_mod
-  !> @{
-  public :: horiz_interp_conserve_init
-  public :: horiz_interp_conserve_new, horiz_interp_conserve, horiz_interp_conserve_del
-
-  integer :: pe, root_pe
-  !-----------------------------------------------------------------------
-  ! Include variable "version" to be written to log file.
-#include<file_version.h>
-  logical            :: module_is_initialized = .FALSE.
-
-  logical         :: great_circle_algorithm = .false.
-
-contains
-
-  !#######################################################################
-
-  !> Writes version number to logfile.
-  subroutine horiz_interp_conserve_init
-
-    if(module_is_initialized) return
-    call write_version_number("HORIZ_INTERP_CONSERVE_MOD", version)
-
-    great_circle_algorithm = get_great_circle_algorithm()
-
-    module_is_initialized = .true.
-
-  end subroutine horiz_interp_conserve_init
-
-  !#######################################################################
-
-  subroutine horiz_interp_conserve_new_1dx1d ( Interp, lon_in, lat_in, lon_out, lat_out, verbose)
+!> @addtogroup horiz_interp_conserve_mod
+!> @{
+subroutine HORIZ_INTERP_CONSERVE_NEW_1DX1D_ ( Interp, lon_in, lat_in, lon_out, lat_out, verbose)
     type(horiz_interp_type), intent(inout) :: Interp
-    real, intent(in),       dimension(:)   :: lon_in , lat_in
-    real, intent(in),       dimension(:)   :: lon_out, lat_out
+    real(FMS_HI_KIND_), intent(in),       dimension(:)   :: lon_in , lat_in
+    real(FMS_HI_KIND_), intent(in),       dimension(:)   :: lon_out, lat_out
     integer, intent(in),       optional    :: verbose
 
     !-----------------------------------------------------------------------
-    real, dimension(size(lat_out(:))-1,2) :: sph
-    real, dimension(size(lon_out(:))-1,2) :: theta
-    real, dimension(size(lat_in(:)))      :: slat_in
-    real, dimension(size(lon_in(:))-1)    :: dlon_in
-    real, dimension(size(lat_in(:))-1)    :: dsph_in
-    real, dimension(size(lon_out(:))-1)   :: dlon_out
-    real, dimension(size(lat_out(:))-1)   :: dsph_out
+    real(FMS_HI_KIND_), dimension(size(lat_out(:))-1,2) :: sph
+    real(FMS_HI_KIND_), dimension(size(lon_out(:))-1,2) :: theta
+    real(FMS_HI_KIND_), dimension(size(lat_in(:)))      :: slat_in
+    real(FMS_HI_KIND_), dimension(size(lon_in(:))-1)    :: dlon_in
+    real(FMS_HI_KIND_), dimension(size(lat_in(:))-1)    :: dsph_in
+    real(FMS_HI_KIND_), dimension(size(lon_out(:))-1)   :: dlon_out
+    real(FMS_HI_KIND_), dimension(size(lat_out(:))-1)   :: dsph_out
     real    :: blon, fac, hpi, tpi, eps
     integer :: num_iters = 4
```

Comment:
integer :: num_iters=4

and later on num_iters=4

Reply:
this can be a parameter